lib/publiccloud/azure.pm

```diff
@@ -26,40 +30,53 @@ sub init {
           . $self->key_secret . ' -t ' . $self->tenantid);
 }
 
-sub filename_to_resgroup {
+sub resource_exist {
+    my ($self) = @_;
+    my $output = script_output(q(az group list --query "[?name=='openqa-upload']"));
```

Comment:
and in case of any not related failure like : 
- azure not accessable ( no internet in SUSE or Microsoft ) 
- we still not logged in 
- account permissions issue 
etc. we will still get this as just like `resource not exists` , right ?

Reply:
if there is a problem like this, the method script_output will die (return code !=0) and we quit the test. So, I think it's still safe.